btcutil/go.mod

```diff
@@ -4,8 +4,10 @@ go 1.16
 
 require (
 	github.com/aead/siphash v1.0.1
-	github.com/btcsuite/btcd v0.20.1-beta
+	github.com/btcsuite/btcd v0.22.0-beta
```

Comment:
The `v0.22.0-beta` version still pulls in `btcsuite/btcutil`, that's why you end up with `btcsuite/btcutil` still being in the top level `go.sum`. If you change this to `v0.22.0-beta.0.20220111032746-97732e52810c`, it goes away.
This is because `replace` directives are ignored in transitive dependencies. So the `replace` directive in this file is ignored when go mod is looking at the top level `btcd` package. Perhaps we should try to get rid of this circular replace directive soon to avoid issues like this?

Reply:
IIRC, once we tag a _new_ version of `btcd` after that, the dep hash goes away?

Reply:
Ah, ok I see now, I can add that. 

> Perhaps we should try to get rid of this circular replace directive soon to avoid issues like this?

IMO the main matter is if we want to have `btcutil` be a proper sub-module or not. The `btcutil/psbt` module also exists. I think as long as we make more sub-modules we'll have this problem until each package is isolated as its own module.

Reply:
The main packages that `btcutil` uses are: `wire`, `chainhash`, `btcec`, and `txscript` (FWIW).